packages/grid/index.ts

```diff
@@ -0,0 +1,209 @@
+import {
+ StarryUICellDimensionsTrait,
+ StarryUICellRenderTrait,
+ StarryUITrait,
+ StarryUITraitConfig,
+ applyTraits,
+ starryComponent,
+} from '@starryui/traits'
+
+export const defaultConfigConfig: StarryUITraitConfig = { themeFacet: 'grid' }
+
+// export const grid = starryComponent<HTMLElement>(function (
+//  traits: StarryUITrait[]
+// ) {
+//  return function (config?: StarryUITraitConfig) {
+//   const elem = document.createElement(config?.tagName ?? 'div')
+//   applyTraits(elem, traits, Object.assign({}, defaultConfigConfig, config))
+//   return elem
+//  }
+// })
+
+const defaultCellDimensionsTrait: StarryUICellDimensionsTrait = {
+ type: 'cellDimensions',
+ height: 120,
+ width: 120,
+}
+const defaultCellRenderTrait: StarryUICellRenderTrait = {
+ type: 'cellRender',
+ cellRender(x, y) {
+  const info = document.createElement('span')
+  info.textContent = `${x}, ${y}`
+  return info
+ },
+}
+
+export const grid = starryComponent<HTMLElement>(function (
+ traits: StarryUITrait[]
+) {
+ const { height, width } =
+  traits.find((x) => x.type === 'cellRender') ?? defaultCellDimensionsTrait
```

Comment:
This should be `cellDimensions` trait not `cellRender` - this will cause an error.

Reply:
Done